### PR TITLE
Fix service sync reject ACL name matching

### PR DESCRIPTION
**Pull request by trozet**
The generateACLName function was including backslash escapes for the ACL
name for use in OVN commands that have trouble with literal ":". That
was causing a mismatch when services were syncing because the name
actually returned from an OVN command does not include any backslashes
so the names would not match.

Signed-off-by: Tim Rozet <trozet@redhat.com>

### CODE REVIEW

**Comment by trozet** [comment]
@danwinship

File: go-controller/pkg/ovn/loadbalancer.go [hunk 1]
```diff
@@ -164,9 +164,9 @@ func (ovn *Controller) getLogicalSwitchesForLoadBalancer(lb string) ([]string, e
 // TODO: Add unittest for function.
 func (ovn *Controller) generateACLName(lb string, sourceIP string, sourcePort int32) string {
 	aclName := fmt.Sprintf("%s-%s:%d", lb, sourceIP, sourcePort)
-	aclName = strings.ReplaceAll(aclName, ":", "\\:")
+	aclNameTmp := strings.ReplaceAll(aclName, ":", "\\:")
 	// ACL names are limited to 63 characters
-	if len(aclName) > 63 {
+	if len(aclNameTmp) > 63 {
```
Comment (danwinship) [inline go-controller/pkg/ovn/loadbalancer.go:169]: is it actually the escaped version that has to be <= 63 characters or the unescaped version?

**Comment by trozet** [comment]
@Billy99

**Review by danwinship — COMMENTED** [review]
so wait... this isn't just shell escaping? OVN requires you to escape it when you pass it in, but it doesn't escape it when it passes it out?

**Comment by danwinship** [comment]
(basically lgtm. feel free to merge if the length check there is right)

File: go-controller/pkg/ovn/loadbalancer.go (hunk 1 above)
Comment (trozet) [reply go-controller/pkg/ovn/loadbalancer.go:169]: @Billy99 added this so I'll let him respond.

**Comment by trozet** [comment]
> so wait... this isn't just shell escaping? OVN requires you to escape it when you pass it in, but it doesn't escape it when it passes it out?

right it looks like it happens when you use a k=v format:
[root@ip-10-0-215-110 ~]# ovn-nbctl --no-leader-only  --columns=name,_uuid --format=json find acl name=84ffba22-20a9-42ad-a8e4-5ad7785e0890-172.30.0.1:443
ovn-nbctl: 84ffba22-20a9-42ad-a8e4-5ad7785e0890-172.30.0.1:443: unexpected ":" parsing set of up to 1 strings


[root@ip-10-0-215-110 ~]# ovn-nbctl --no-leader-only  --columns=name,_uuid --format=json find acl name=84ffba22-20a9-42ad-a8e4-5ad7785e0890-172.30.0.1\\:443 
{"data":[["84ffba22-20a9-42ad-a8e4-5ad7785e0890-172.30.0.1:443",["uuid","463ccb51-4f5c-42ae-bc6b-c557cc55f51e"]]],"headings":["name","_uuid"]}


[root@ip-10-0-215-110 ~]# ovn-nbctl --no-leader-only --no-headings --columns=name list acl 463ccb51-4f5c-42ae-bc6b-c557cc55f51e
"84ffba22-20a9-42ad-a8e4-5ad7785e0890-172.30.0.1:443"

File: go-controller/pkg/ovn/loadbalancer.go (hunk 1 above)
Comment (trozet) [reply go-controller/pkg/ovn/loadbalancer.go:169]: the esacpes don't matter for the total length being <= 63 so will fix this

**Comment by danwinship** [comment]
looks good

**Review by Billy99 — COMMENTED** [review]
No longer needed. You can remove the check for IP type as well now.

**Comment by Billy99** [comment]
@trozet  Sorry, last comment didn't have code context (I guess because I commented on an unchanged line). In generateACLName(), it the len is greater than 63, there is a calculation of ipPortLen, which includes a 14 len for the \\. That 14 can be removed, which means the IP type check can also be removed.

**Pull request merged**